pkg/antctl/raw/multicluster/join/join.go

```diff
@@ -0,0 +1,368 @@
+		fmt.Fprintf(cmd.OutOrStdout(), "Failed to get secret \"%s\" in Namespace %s: %v\n", options.LeaderClusterOpt.Secret, options.MemberClusterOpt.Namespace, err)
+		return err
+	}
+
+	crtData, token, err := commonarea.GetSecretCACrtAndToken(secret)
+	if err != nil {
+		return err
+	}
+	config, err := clientcmd.BuildConfigFromFlags(options.LeaderClusterOpt.Server, "")
+	if err != nil {
+		return err
+	}
+	config.BearerToken = string(token)
+	config.CAData = crtData
+
+	remoteClient, err := client.New(config, client.Options{Scheme: multiclusterscheme.Scheme})
+	if err != nil {
+		return err
+	}
+
+	memberClusterAnnounce := "member-announce-from-" + options.MemberClusterOpt.ClusterID
+	fmt.Fprintf(cmd.OutOrStdout(), "Waiting for MemberClusterAnnounce \"%s\" in Namespace %s in leader cluster\n", memberClusterAnnounce, options.LeaderClusterOpt.Namespace)
+	if err := waitForMemberCluserAnnounceReady(remoteClient, memberClusterAnnounce, options.LeaderClusterOpt.Namespace); err != nil {
+		fmt.Fprintf(cmd.OutOrStdout(), "Failed to wait for MemberClusterAnnounce \"%s\" in Namespace %s in leader cluster: %v\n", memberClusterAnnounce, options.LeaderClusterOpt.Namespace, err)
+		return err
+	}
+
+	if err := waitForLeaderClusterSetReady(remoteClient, options.LeaderClusterOpt.ClusterSet, options.LeaderClusterOpt.Namespace, options.MemberClusterOpt.ClusterID); err != nil {
+		fmt.Fprintf(cmd.OutOrStdout(), "Failed to wait for ClusterSet \"%s\" in Namespace %s in leader cluster: %v\n", options.LeaderClusterOpt.ClusterSet, options.LeaderClusterOpt.Namespace, err)
+		return err
+	}
+
+	return nil
+}
+
+func waitForMemberCluserAnnounceReady(client client.Client, memberClusterAnnounce string, namespace string) error {
+	return wait.PollImmediate(
+		1*time.Second,
+		30*time.Second,
+		func() (bool, error) {
+			announce := &multiclusterv1alpha1.MemberClusterAnnounce{}
+			if err := client.Get(context.TODO(), types.NamespacedName{Name: memberClusterAnnounce, Namespace: namespace}, announce); err != nil {
+				if apierrors.IsNotFound(err) {
+					return false, nil
+				}
+				return false, err
+			}
+			return announce.DeletionTimestamp == nil, nil
+		})
+}
+
+func waitForLeaderClusterSetReady(client client.Client, name string, namespace string, memberClusterID string) error {
+	return wait.PollImmediate(
+		1*time.Second,
+		3*time.Minute,
+		func() (bool, error) {
+			clusterSet := &multiclusterv1alpha1.ClusterSet{}
+			if err := client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, clusterSet); err != nil {
+				if apierrors.IsNotFound(err) {
+					return false, nil
+				}
+				return false, err
+			}
+
+			for _, member := range clusterSet.Status.ClusterStatuses {
+				if member.ClusterID == memberClusterID {
+					for _, cond := range member.Conditions {
+						if cond.Reason == "Connected" {
+							return cond.Status == "True", nil
+						}
+					}
+				}
```

Comment:
You may add `break` here to stop the loop after you find the right member.

Reply:
break added.